### ISSUE TITLE
"ERROR: cache lookup failed for type 2" with a function the first time it run.

### ISSUE THREAD

**Issue by JMLessard** [pgaudit/pgaudit#65]
I compiled posgreSQL 11 for windows with MSYS2 64bit (includes mingw64) following the procedure outline in
https://www.cybertec-postgresql.com/en/building-postgresql-with-msys2-and-mingw-under-windows/

```
select version();
                                                               version
-------------------------------------------------------------------------------------------------------------------------------------
 PostgreSQL 11.1 on x86_64-w64-mingw32, compiled by x86_64-w64-mingw32-gcc.exe (Rev1, Built by MSYS2 project) 8.2.1 20181207, 64-bit

```
When enabling pgaudit, I got the "ERROR:  cache lookup failed for type 2" when a function is run for the first time in a session.
The function run successfully the second time.
If pgaudit is not enabled, the function run without issue.

shared_preload_libraries = 'pgaudit'
pgaudit.log = 'all'
pgaudit.log_catalog = on
pgaudit.log_parameter = on

pgaudit 1.3 is installed (the PG11 compatible version)
```
select * from pg_extension;
      extname       | extowner | extnamespace | extrelocatable | extversion | extconfig | extcondition
--------------------+----------+--------------+----------------+------------+-----------+--------------
 pgaudit            |       10 |        16432 | t              | 1.3        | (NULL)    | (NULL)

```
Please find a the test function to reproduce the issue at https://www.postgresql.org/message-id/flat/23950.1544819190%40sss.pgh.pa.us#64ab3d47fff143ea9b64917a79b6d284
You do not need any FK in the test schema to reproduce the problem.

**Comment by JMLessard** [comment]
Hi, did you have sometime to look at the issue?
Are you able to reproduce it?

My best regards for the new year,

**Comment by JMLessard** [comment]
Hi, did you have sometime to look at the issue?
Are you able to reproduce it?

My best regards for the new year,

**Comment by dwsteele** [comment]
I don't have a Windows machine (or Windows knowledge) to try this on.  I pinged a few people who might know more but they may be busy.

**Comment by jconway** [comment]
I have reproduced this on Linux (using a different query) and will make a pull request with a fix.

**Comment by dwsteele** [comment]
Thanks, @jconway!

**Comment by dwsteele** [comment]
@JMLessard: this has been fixed in b2040b6e6c74718afd41b4ba46243fc31a286cfa and back-patched to the REL11_STABLE branch.  Give it a try and see if it fixes your problem.

**Comment by JMLessard** [comment]
Great!, yes it does fix
Thank you very much
